### ISSUE TITLE
May I ask if there is an official evaluation code for the GOLD algorithm?

### ISSUE THREAD

**Issue by 141forever** [huggingface/trl#4886]
### Reproduction

Please, Thanks.

### System Info

None

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by 141forever** [comment]
1. I would like to know the evaluation method for the CountDown task.
2. I want to know what the training parameters are for the GOLD task.

The settings here: https://huggingface.co/spaces/HuggingFaceH4/on-policy-distillation. 
First, a learning rate of 1e-7 does not allow the model loss to decrease; I have to increase it to around 1e-5.
Second, the results after training for 1000 epochs are completely different from what is reported in the paper, and the loss is even below 0.01.
Below are my training parameters (using 2 H200 GPUs).

training_args = GOLDConfig(
        output_dir="gold-model-llama",
        save_strategy="steps",
        save_steps=500,
        learning_rate=2e-4,
        warmup_ratio=0.05,
        per_device_train_batch_size=16,
        teacher_model_name_or_path=teacher_name,
        teacher_tokenizer_name_or_path=teacher_name,
        bf16=True,
        use_uld_loss=True,
        uld_use_hybrid_loss=True,
        push_to_hub=False,
        report_to=[],
        lr_scheduler_type = 'cosine',
        num_train_epochs=5,
        max_steps=2000,
        logging_steps=10,
        gradient_accumulation_steps=1,
        lmbda = 1.0,
        beta = 0.0,
        uld_crossentropy_weight = 0,
        uld_distillation_weight = 1.0,
    )

**Comment by albertvillanova** [comment]
Hey @141forever, sorry for the slow reply — we're catching up on the backlog. 

GOLD lives in `trl.experimental`; TRL ships the trainer, the `HuggingFaceTB/Countdown-Task-GOLD` dataset, and an example config, but not a standalone CountDown eval harness or a paper-reproduction recipe — those live with the on-policy-distillation work you linked, so I'd point you to the HuggingFaceH4/on-policy-distillation Space and to the trainer author @kashif. On the LR: the `GOLDConfig` default of `1e-7` is intentionally conservative (https://github.com/huggingface/trl/blob/7904dc11cf8049c70e622e9ab3200ee5d9aa044d/trl/experimental/gold/gold_config.py#L168-L169) and much lower than the documented example (`2e-5`), so bumping to `1e-5`–`2e-4` as you did is the right call. And with `uld_crossentropy_weight=0`, `uld_distillation_weight=1.0`, `lmbda=1.0`, `beta=0.0` you're running pure on-policy distillation, so a tiny training loss (<0.01) is expected and isn't directly comparable to the paper's task metric — to match the paper you'd run the task-level CountDown accuracy eval, not the loss. 

Closing since this isn't a TRL bug, but happy to keep helping if you re-open with specifics or ping [@kashif](https://huggingface.slack.com/team/U048THPQ741) in-thread.

**Comment by kashif** [comment]
Hey Andy! Perhaps check the xtokens paper for competitive baselines to compare new methods.
